controllers/k8ssandra/cleanup.go

```diff
@@ -357,3 +366,69 @@ func (r *K8ssandraClusterReconciler) deleteK8ssandraConfigMaps(
 	}
 	return
 }
+
+func (r *K8ssandraClusterReconciler) deleteServices(
+	ctx context.Context,
+	kc *k8ssandraapi.K8ssandraCluster,
+	dcTemplate k8ssandraapi.CassandraDatacenterTemplate,
+	namespace string,
+	remoteClient client.Client,
+	kcLogger logr.Logger,
+) (hasErrors bool) {
+	selector := k8ssandralabels.CleanedUpByLabels(client.ObjectKey{Namespace: kc.Namespace, Name: kc.Name})
+	options := client.ListOptions{
+		Namespace:     namespace,
+		LabelSelector: labels.SelectorFromSet(selector),
+	}
+	serviceList := &corev1.ServiceList{}
```

Comment:
Suggestion: A better option here would probably be `DeleteAllOf`, using the labels in the LabelSelector. I am happy for you to add this to the refactoring ticket you already have open to fix these functions up and combine them into a single generic function.